crates/bevy_ui/src/layout/mod.rs

```diff
@@ -5,44 +5,69 @@ use crate::{ContentSize, Node, Style, UiScale};
 use bevy_ecs::{
     change_detection::DetectChanges,
     entity::Entity,
-    event::EventReader,
+    prelude::Component,
     query::{With, Without},
+    reflect::ReflectComponent,
     removal_detection::RemovedComponents,
-    system::{Query, Res, ResMut, Resource},
+    system::{ParamSet, Query, Res, ResMut, Resource},
     world::Ref,
 };
 use bevy_hierarchy::{Children, Parent};
 use bevy_log::warn;
 use bevy_math::Vec2;
+use bevy_reflect::{std_traits::ReflectDefault, Reflect};
 use bevy_transform::components::Transform;
 use bevy_utils::HashMap;
-use bevy_window::{PrimaryWindow, Window, WindowResolution, WindowScaleFactorChanged};
+use bevy_window::Window;
 use std::fmt;
 use taffy::{prelude::Size, style_helpers::TaffyMaxContent, Taffy};
 
+/// Marks an entity as `UI root entity` with an associated root Taffy node and holds the resolution and scale factor information necessary to compute a UI layout.
+#[derive(Component, Debug, Reflect)]
+#[reflect(Component, Default)]
 pub struct LayoutContext {
-    pub scale_factor: f64,
-    pub physical_size: Vec2,
-    pub min_size: f32,
-    pub max_size: f32,
+    /// The size of the root node in the layout tree.
+    ///
+    /// Should match the size of the output window in physical pixels of the display device.
```

Comment:
Is it a hard requirement that the root node covers the entire window? Or is it just the most common use case?

Reply:
Reading below seems to suggest yes. Then I'd rephrase to be more assertive than "should".

```suggestion
    /// Set automatically during layout computation to the size of the output window in physical pixels of the display device.
```

Reply:
Yeah at the moment `ui_system_layout` sets the values for `LayoutContext` automatically when it's a component of a `Window` entity.